compiler/src/iree/compiler/Codegen/Dialect/CPU/IR/IREECPUAttrs.cpp

```diff
@@ -207,6 +207,54 @@ Attribute LoweringConfigAttr::getTilingLevelAttr(unsigned level) const {
   return config.get(key);
 }
 
+constexpr std::array vectorTilingLevels{TilingLevel::VectorCommonParallelTiles,
+                                        TilingLevel::VectorReductionTiles,
+                                        TilingLevel::VectorInnerParallelTiles};
+
+std::optional<SmallVector<int64_t>> LoweringConfigAttr::getVectorSizes() const {
+  SmallVector<int64_t> result;
+  for (auto level : vectorTilingLevels) {
+    if (!hasTilingLevel(level)) {
+      continue;
+    }
+    auto attr = cast<IREE::Codegen::LoweringConfigTilingLevelAttr>(
+        getTilingLevelAttr(level));
+    if (result.empty()) {
+      result.resize(attr.getSizes().size(), 0);
+    }
+    for (auto [idx, size] : llvm::enumerate(attr.getSizes())) {
+      if (size == 0) {
+        continue;
+      }
+      if (result[idx] != 0) {
+        return std::nullopt;
+      }
+      result[idx] = size;
```

Comment:
I guess at most one of these will be `!= 0`:
* `VectorCommonParallelTiles[idx]`,
* `VectorReductionTiles[idx]`,
* `VectorInnerParallelTiles[idx]`.
?

Reply:
Makes sense. I would just add some comments to document that that's the intended design :)

Reply:
Yes, sounds good to me, thanks!

Reply:
Done, I added the doc to IREECPUAttrs.td.